packages/platform-android/src/commands/runAndroid/index.js

```diff
@@ -146,9 +149,13 @@ function buildApk(gradlew) {
     execFileSync(gradlew, ['build', '-x', 'lint'], {
       stdio: [process.stdin, process.stdout, process.stderr],
     });
+
+    return true;
   } catch (e) {
     logger.error('Could not build the app, read the error above for details.');
```

Comment:
We could avoid returning a boolean from this function and verifying it inside `runOnSpecificDevice` by throwing the error instead of logging it. Like:
```js
throw new CLIError('Could not build the app, read the error above for details', e)
```

Reply:
Actually, we could do it even better.
```js
throw new CLIError('There was a problem building the app', originalErrorMessage)
```
and user would see the original error when running with `--verbose` flag.

We would have to stop piping stdout/stderr tho.

Reply:
@grabbou try it locally. The code inside CLIError only shows first 2 lines because it does `.split(0, 2)` (I missed that during code review). Feel free to send a fix separately from this PR :)

Reply:
How come: https://github.com/react-native-community/cli/blob/master/packages/tools/src/errors.js#L18 - the `split` should not run when string is passed, not an object.

Reply:
Anyway, we could not pipe `stderr` to `process.stderr` and simply pass it as a second argument to CLIError to show the huge error output in the verbose mode only (actually, we could do the same for stdout too, but this is getting out of scope for this PR)

Reply:
Oh, yea, it will work with a string message, you're right. I thought you were passing whole error (then where the stripping comes in place):

```
BUILD FAILED in 0s
error Failed to build the app with gradlew.
debug Error: Command failed: ./gradlew build -x lint
    at checkExecSyncError (child_process.js:616:11)
```
indstead of:
```
BUILD FAILED in 0s
error Failed to build the app with gradlew.
debug Error: Failed to build the app with gradlew.
    at buildApk (/Users/thymikee/Projects/oss/react-native-cli/packages/platform-android/build/commands/runAndroid/index.js:168:11)
    at runOnSpecificDevice (/Users/thymikee/Projects/oss/react-native-cli/packages/platform-android/build/commands/runAndroid/index.js:149:7)
    at buildAndRun (/Users/thymikee/Projects/oss/react-native-cli/packages/platform-android/build/commands/runAndroid/index.js:135:14)
    at then.result (/Users/thymikee/Projects/oss/react-native-cli/packages/platform-android/build/commands/runAndroid/index.js:102:12)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Reply:
Thanks for your feedbacks. I really agree with using exception. I didn't know the changes in the 2.0 branch.